wgpu-core/src/device/global.rs

```diff
@@ -58,10 +58,39 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
 
             hal_caps.formats.sort_by_key(|f| !f.is_srgb());
 
+            let mut usages = wgt::TextureUsages::empty();
+            usages.set(
+                wgt::TextureUsages::COPY_SRC,
+                hal_caps.usage.contains(hal::TextureUses::COPY_SRC),
+            );
+            usages.set(
+                wgt::TextureUsages::COPY_DST,
+                hal_caps.usage.contains(hal::TextureUses::COPY_DST),
+            );
+            usages.set(
+                wgt::TextureUsages::TEXTURE_BINDING,
+                hal_caps.usage.contains(hal::TextureUses::RESOURCE),
+            );
+            usages.set(
+                wgt::TextureUsages::STORAGE_BINDING,
+                hal_caps.usage.contains(
+                    hal::TextureUses::STORAGE_READ | hal::TextureUses::STORAGE_READ_WRITE,
+                ),
+            );
+            usages.set(
+                wgt::TextureUsages::RENDER_ATTACHMENT,
+                hal_caps.usage.contains(hal::TextureUses::COLOR_TARGET)
+                    | hal_caps.usage.contains(
+                        hal::TextureUses::DEPTH_STENCIL_READ
+                            | hal::TextureUses::DEPTH_STENCIL_WRITE,
+                    ),
+            );
```

Comment:
there's a `map_texture_usage` on `conv.rs` that you should be able to use for this

Reply:
That one is the inverse mapping unfortunately. It goes TextureUsages -> TextureUses, but we need TextureUses -> TextureUsages here. I based this logic on inverting the logic I found there. We could extract all of this to its own function in conv.rs for going in this direction though. Personally I think this would probably be more elegant as From<T> implementations in both directions.

Reply:
oh, you're right ofc, didn't pay enough attention
Yeah let's put that into conv.rs, surprised we didn't have it yet :)